### DISCUSSION TITLE
does hmr also work for all 3 types of content script? (declared statically, declared dynamically, and programmatically injected)

### DISCUSSION THREAD

**Discussion by devhandler** [NekitCorp/chrome-extension-svelte-typescript-boilerplate#24] [Q&A]
sometimes we have to use dynamic or progammatic injection for content script. does this lib also support these 2 types of content script?

**Comment by NekitCorp** [comment]
Hello, yes it looks like all 3 types of content scripts work correctly with HMR:

1. [Inject with static declarations](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#static-declarative)
2. [Inject with dynamic declarations](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#dynamic-declarative)
3. [Inject programmatically](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#programmatic)

You can see an example in [content_scripts](https://github.com/NekitCorp/chrome-extension-svelte-typescript-boilerplate/tree/content_scripts) branch.

https://github.com/NekitCorp/chrome-extension-svelte-typescript-boilerplate/assets/15874982/56a5fb57-e272-4dc3-ae95-30fa3928e968

```ts
import injectProgrammatically from "../content/inject-programmatically?script";
import injectWithDynamicDeclarations from "../content/inject-with-dynamic-declarations?script";

// Inject programmatically
// https://developer.chrome.com/docs/extensions/mv3/content_scripts/#programmatic
chrome.action.onClicked.addListener((tab) => {
    if (tab.id) {
        chrome.scripting.executeScript({
            target: { tabId: tab.id },
            files: [injectProgrammatically],
        });
    }
});

// Inject with dynamic declarations
// https://developer.chrome.com/docs/extensions/mv3/content_scripts/#dynamic-declarative
chrome.scripting
    .registerContentScripts([
        {
            id: "session-script",
            js: [injectWithDynamicDeclarations],
            persistAcrossSessions: false,
            matches: ["https://*/*"],
            runAt: "document_start",
        },
    ])
    .then(() => console.log("registration complete"))
    .catch((err) => console.warn("unexpected error", err));`
```
